Makefile

```diff
@@ -454,6 +454,11 @@ release: clean-release ## Builds and push container images using the latest git
 	$(MAKE) set-manifest-pull-policy PULL_POLICY=IfNotPresent TARGET_RESOURCE="./bootstrap/kubeadm/config/manager/manager_pull_policy.yaml"
 	$(MAKE) set-manifest-pull-policy PULL_POLICY=IfNotPresent TARGET_RESOURCE="./controlplane/kubeadm/config/manager/manager_pull_policy.yaml"
 	$(MAKE) release-manifests
+	# Release CAPD components and add them to the release dir
+	$(MAKE) -C test/infrastructure/docker/ release
+	cp test/infrastructure/docker/out/infrastructure-components.yaml $(RELEASE_DIR)/infrastructure-components.yaml
```

Comment:
Can we rename this to something `infrastructure-dev-components` or something similar?

One other thing to point out is that CAPD doesn't get go module tags, I assume this is ok for now?

Reply:
> Can we rename this to something infrastructure-dev-components or something similar?

I was thinking on the same lines, but then rolled back because I can't do the same for cluster templates unless there is an exception for the quick start for the docker provider (it requires to add `--flavor` dev to `clusterctl get cluster`)
But I'm ok with both options, just let me know which one you prefer:
a) add `-dev` + the document the exception
b) leave the current name given that that CAPD is not for production is already documented everywhere

> One other thing to point out is that CAPD doesn't get go module tags, I assume this is ok for now?

CAPD image is generated using the same tag used by CAPI, KCP, CABPK, so I'm using this tag for the CAPD release manifest as well